From 30dee41f48dbe5d0330044d0103f46a483e8b03f Mon Sep 17 00:00:00 2001
From: Jianxu Zheng <jian.xu.zheng@intel.com>
Date: Tue, 2 Dec 2014 14:23:32 +0800
Subject: [PATCH 164/164] atomisp: add ioctl for sensors to avoid lowering fps

In depth mode, the lowered framerate is causing frame synch problems
with the depth cameras. This patch introduces an ioctl to notify
the sensor driver that the camera HAL is running in depth mode and
in that case will prevent lowering the framerate due
to too high exposure values.

Depends on: Ic629bc7e2a4c05f9a5b83584af4204907d1b2222
Depends on: I7322a5716332a756edffc200399954fd9f840580

Change-Id: I788ed08fe7609131b27e1793a54941f3e43ec2f7
Chang-On: https://jira01.devtools.intel.com/browse/IMINAN-7421
Signed-off-by: Soren Friis <soren.friis@intel.com>
Signed-off-by: Jianxu Zheng <jian.xu.zheng@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/31007
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/atomisp_driver/atomisp_ioctl.c    |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 5c13bfd..02fdf03 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -231,6 +231,15 @@ static struct v4l2_queryctrl ci_v4l2_controls[] = {
 		.default_value = 0,
 	},
 	{
+		.id = V4L2_CID_EXPOSURE_AUTO_PRIORITY,
+		.type = V4L2_CTRL_TYPE_INTEGER,
+		.name = "Exposure auto priority",
+		.minimum = V4L2_EXPOSURE_AUTO,
+		.maximum = V4L2_EXPOSURE_APERTURE_PRIORITY,
+		.step = 1,
+		.default_value = V4L2_EXPOSURE_AUTO,
+	},
+	{
 		.id = V4L2_CID_SCENE_MODE,
 		.type = V4L2_CTRL_TYPE_INTEGER,
 		.name = "scene mode",
@@ -2153,6 +2162,7 @@ static int atomisp_s_ctrl(struct file *file, void *fh,
 	switch (control->id) {
 	case V4L2_CID_AUTO_N_PRESET_WHITE_BALANCE:
 	case V4L2_CID_EXPOSURE:
+	case V4L2_CID_EXPOSURE_AUTO_PRIORITY:
 	case V4L2_CID_SCENE_MODE:
 	case V4L2_CID_ISO_SENSITIVITY:
 	case V4L2_CID_ISO_SENSITIVITY_AUTO:
-- 
1.7.9.5

